File: T1/codigos2223/Repaso/src/ejerciciocartas/Carta.java
package ejerciciocartas;

public class Carta {

    private char palo;
    private int valor;

    public Carta (int valor, char palo){
        this.valor = valor;
        this.palo = palo;
    }

    public void mostrarDatos(){
        System.out.print(getValor()+""+getPalo()+", ");
    }

    public char getPalo() {
        return palo;
    }

    public void setPalo(char palo) {
        this.palo = palo;
    }

    public int getValor() {
        return valor;
    }

    public void setValor(int valor) {
        this.valor = valor;
    }
}

File: Tema4/codigos_ret/dom/carta.js
class Carta {
  // atributos
  imagen;
  valor;
  representacion;
  // constructores
  constructor(representacion) {
    this.imagen = "./images/" + representacion + ".png";
    // 1C
    // 10T
    this.valor = representacion.substring(0, representacion.length - 1);
    this.representacion = representacion;
  }
  // metodos
  mostrarDatos() {
    console.log(this.imagen);
    console.log(this.valor);
    console.log(this.representacion);
  }

  setValor(valor) {
    this.valor = valor;
  }

  getValor() {
    return this.valor;
  }
}
